### PR TITLE
Fix: Issue 597

### PR DESCRIPTION
**Pull request by HoshinoTented**
as title.

### CODE REVIEW

**Comment by HoshinoTented** [comment]
The error message to this code:

```aya
open data Nat | O | S Nat
def bad Nat : Nat | S S O => O | _ => O
```

is now:

<details>
<summary>Aya said:</summary>

```
Failed with `class org.aya.tyck.pat.PatternProblem$InsufficientPattern`: In file main.aya:2:22 ->

  1 | open data Nat | O | S Nat
  2 | def bad Nat : Nat | S S O => O | _ => O
                            ^^

Error: There is no pattern for the parameter
         (_5 : Nat)

at (48-48) [2,22-2,22]
java.lang.AssertionError: Failed with `class org.aya.tyck.pat.PatternProblem$InsufficientPattern`: In file main.aya:2:22 ->

  1 | open data Nat | O | S Nat
  2 | def bad Nat : Nat | S S O => O | _ => O
                            ^^

Error: There is no pattern for the parameter
         (_5 : Nat)
```

</details>

File: base/src/main/java/org/aya/tyck/pat/PatTycker.java [hunk 1]
```diff
@@ -288,20 +289,38 @@ private LhsResult checkLhs(Pattern.Clause match, Def.Signature signature) {
...
   }
 
+  /**
+   * @param outerPattern null if visiting the whole pattern (like `A, x, ctor a b`). This is only used for error reporting.
+   *                     For now, {@param outerPattern} is used when {@param sig} is not empty
+   *                     but {@param stream} is empty, it is possible when matching parameters of Ctor.
+   */
   public @NotNull Tuple2<SeqView<Pat>, Term>
-  visitPatterns(Def.Signature sig, SeqView<Pattern> stream) {
+  visitPatterns(@NotNull Def.Signature sig, @NotNull SeqView<Pattern> stream, @Nullable Pattern outerPattern) {
```
Comment (HoshinoTented) [inline base/src/main/java/org/aya/tyck/pat/PatTycker.java:298]: In fact, I don't think that adding a new argument is good.

Comment (ice1000) [reply base/src/main/java/org/aya/tyck/pat/PatTycker.java:302]: It's ok

**Review by ice1000 — DISMISSED** [review]
bors d=HoshinoTented

Merge if you think it's ready.

**Comment by ice1000** [comment]
Merge?

**Pull request merged**